src_multispec/MassFluxUtil.cpp

```diff
@@ -24,9 +24,19 @@ void ComputeMolconcMolmtot(const MultiFab& rho_in,
 
         amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
         {
+
+            GpuArray<Real, MAX_SPECIES> RhoN;
+            GpuArray<Real, MAX_SPECIES> MolarConcN;
+
+            for (int n=0; n<nspecies; ++n ){
+                RhoN[n] = rhotot(i,j,k,n);
+                MolarConcN[n] = molarconc(i,j,k,n);
```

Comment:
Since ComputeMolconcMolmtotLocal is computing molar concentrations, you should delete this line and then *after* the call to ComputeMolconcMolmtotLocal you need something like "molarconc(i,j,k,n) = MolarConcN[n]"